src/parameterFactory.js

```diff
@@ -0,0 +1,124 @@
+import {
+  DateInput,
+  EmailInput,
+  NullableBooleanInput,
+  NumberInput,
+  TextInput,
+} from 'react-admin';
+import React from 'react';
+
+function guessType(parameter, fields) {
+  if (parameter.variable.match(/.*\[exists\]/i) !== null) {
+    return 'nullableBoolean';
+  }
+
+  if (
+    parameter.variable.match(/.*\[after\]/i) !== null ||
+    parameter.variable.match(/.*\[before\]/i) !== null ||
+    parameter.variable.match(/.*\[strictly_after\]/i) !== null ||
+    parameter.variable.match(/.*\[strictly_after\]/i) !== null
+  ) {
+    return 'date';
+  }
+
+  if (parameter.variable.match(/.*\[between\]/i) !== null) {
+    return 'between';
+  }
+
+  return guessTypeFromResource(parameter, fields);
+}
+
+function guessTypeFromResource(parameter, fields) {
+  const type = fields
+    .filter(field => {
+      return field['name'] === parameter.variable;
+    })
+    .map(field => {
+      switch (field.id) {
+        case 'http://schema.org/email':
+          return 'email';
+        default:
+      }
+
+      switch (field.range) {
+        case 'http://www.w3.org/2001/XMLSchema#integer':
+        case 'http://www.w3.org/2001/XMLSchema#float':
+          return 'number';
+
+        case 'http://www.w3.org/2001/XMLSchema#boolean':
+          return 'nullableBoolean';
+
+        default:
+      }
+
+      return 'text';
+    });
+
+  return type.length > 0 ? type[0] : 'text';
+}
+
+export default (parameter, fields, options) => {
+  if (
+    // List filters are discarded because there is no built-in filter component in react-admin that can handle this case.
+    parameter.variable.match(/.*\[\]/i) ||
```

Comment:
Maybe could we introduce a `api-platform` flag allowing to disable all these "vendor specific" tweaks?